### PR TITLE
Add an event filter with the resource.DesiredStateChanged predicate to filter status updates out

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

This PR adds an event filter using the [resource.DesiredStateChanged](https://github.com/crossplane/crossplane-runtime/blob/27a547d7e4fc71f4ecf8cbe155a820a283ae6c0e/pkg/resource/predicates.go#L170) to the generated resource controllers so that updates to only labels, annotations (Crossplane's create pending/failed annotations will be ignored) and to the `spec` will result in reconcile requests.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Tested with https://github.com/upbound/upjet/pull/231.

### CODE REVIEW

File: config/kendra/config.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kendra
+
+import (
+	"github.com/upbound/upjet/pkg/config"
+)
+
+// Configure adds configurations for kafka group.
```
Comment (sergenyalcin) [inline config/kendra/config.go:21]:
```suggestion
// Configure adds configurations for kendra group.
```

File: config/medialive/config.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package medialive
+
+import (
+	"github.com/upbound/upjet/pkg/config"
+)
+
+// Configure adds configurations for kafka group.
```
Comment (sergenyalcin) [inline config/medialive/config.go:21]:
```suggestion
// Configure adds configurations for medialive group.
```

**Review by sergenyalcin — APPROVED** [review]
Thanks @ulucinar I left two comments that are related to copy-paste.

File: config/kendra/config.go (hunk 1 above)
Comment (ulucinar) [reply config/kendra/config.go]: Thanks @sergenyalcin, done.

File: config/medialive/config.go (hunk 2 above)
Comment (ulucinar) [reply config/medialive/config.go]: Done.

**Pull request merged**